management/server/policy.go

```diff
@@ -363,7 +373,9 @@ func (am *DefaultAccountManager) SavePolicy(ctx context.Context, accountID, user
 	}
 	am.StoreEvent(ctx, userID, policy.ID, accountID, action, policy.EventMeta())
 
-	am.updateAccountPeers(ctx, account)
+	if !policy.AreAllRuleGroupsEmpty() {
+		am.updateAccountPeers(ctx, account)
+	}
```

Comment:
This checks only after the policy is updated, it does not consider if before the updates the groups have not been empty